src/Transform/ONNX/ConstProp.td

```diff
@@ -43,6 +43,8 @@ include "src/Dialect/ONNX/ONNX.td"
 
 def IsNoneType : Constraint<CPred<"isa<NoneType>(($_self).getType())">>;
 
+def IsNotStringType : Constraint<CPred<"! isa<ONNXStringType>(($_self).getType().cast<ShapedType>().getElementType())">>;
```

Comment:
suggestion: since it doesn't work on complex types either, how about we change it to test `..getElementType().isIntOrFloatType()`?

and we could name it HasIntegerOrFloatElementType to follow the convention of HasIntegerElementType below